### PR TITLE
Connect directly to content api for content notification expiry

### PR DESCRIPTION
**Pull request by davidfurey**
In preparation for replacing content notifications with a lambda

### CODE REVIEW

**Comment by davidfurey** [comment]
Depends on https://github.com/guardian/mobile-platform/pull/460

File: common/src/main/scala/auditor/package.scala [hunk 1]
```diff
@@ -1,26 +1,63 @@
...
+          case false => Some(topic)
+        }
+      }.map(_.flatten)
+    }
+
+    def isLiveBloggingNow(id: String)(implicit ec: ExecutionContext): Future[Boolean] = {
+      val url = s"${config.url}/$id?show-fields=liveBloggingNow&api-key=${config.apiKey}"
+      wsClient.url(url).get().map { response =>
+        val liveBloggingNowField = response.json \ "response" \ "content" \ "fields" \ "liveBloggingNow"
+        liveBloggingNowField.validate[String].asOpt.flatMap(s => Try(s.toBoolean).toOption).getOrElse(false)
```
Comment (alexduf) [inline common/src/main/scala/auditor/package.scala]: I think you can use `validateOpt[Boolean].getOrElse(false)` here

File: common/src/main/scala/auditor/package.scala [hunk 2]
```diff
@@ -1,26 +1,63 @@
...
+      } recover { case _ => true }
```
Comment (alexduf) [inline common/src/main/scala/auditor/package.scala]: why defaulting to false above, but recovering to true here, I feel like I missed something?

File: common/src/main/scala/auditor/package.scala [hunk 3]
```diff
@@ -1,26 +1,63 @@
...
+
+    override def expiredTopics(topics: Set[Topic])(implicit ec: ExecutionContext): Future[Set[Topic]] = {
+      val contentTopics = topics.filter(_.`type` == TopicTypes.Content)
+
+      Future.traverse(contentTopics) { topic =>
+        isLiveBloggingNow(topic.name).map {
+          case true => None
+          case false => Some(topic)
+        }
+      }.map(_.flatten)
```
Comment (DiegoVazquezNanini) [inline common/src/main/scala/auditor/package.scala]: flatMap instead?

Comment (davidfurey) [inline common/src/main/scala/auditor/package.scala]: I don't think that will work

File: common/src/main/scala/auditor/package.scala (hunk 1 above)
Comment (davidfurey) [inline common/src/main/scala/auditor/package.scala]: The "liveBloggingNow" field is a string not a boolean in the json, so I don't think it would parse

**Comment by DiegoVazquezNanini** [comment]
:+1: Looks good to me, but maybe someone else should take a second look.
Cheers

File: common/src/main/scala/auditor/package.scala (hunk 2 above)
Comment (davidfurey) [inline common/src/main/scala/auditor/package.scala]: If the field does not exist then it isn't a live blog but if we can't connect to content-api we have no idea if it is a live blog or not, so 'true' is a safer guess.

File: common/src/main/scala/auditor/package.scala [hunk 4]
```diff
@@ -1,26 +1,63 @@
...
-import models.Topic
+import models.{Topic, TopicTypes}
+import play.api.Logger
 import play.api.libs.json.Json
+import play.api.libs.ws.WSClient
 
 import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
 
 package object auditor {
```
Comment (davidfurey) [inline common/src/main/scala/auditor/package.scala]: Fair.  I think there is probably scope for a general refactor of this project soon, now that it has gotten a bit larger.

Comment (alexduf) [inline common/src/main/scala/auditor/package.scala]: Yeah I think that's a good idea, starting by putting general package names (my bad :/)

File: common/src/main/scala/auditor/package.scala (hunk 1 above)
Comment (alexduf) [inline common/src/main/scala/auditor/package.scala]: Ah I see

File: common/src/main/scala/auditor/package.scala (hunk 2 above)
Comment (alexduf) [inline common/src/main/scala/auditor/package.scala]: That makes sense then!

**Pull request merged**